security/src/main/java/com/fincity/security/service/ClientPasswordPolicyService.java

```diff
@@ -45,29 +45,40 @@ public class ClientPasswordPolicyService extends
 	private CacheService cacheService;
 
 	private final Set<Character> specialCharacters = Set.of('~', '`', '!', '@', '#', '$', '%', '^', '&', '*', '(', ')',
-			'_', '-', '+', '=', '{', '}', '[', ']', '|', '\\', '/', ':', ';', '\"', '\'', '<', '>', ',', '.', '?');
+	        '_', '-', '+', '=', '{', '}', '[', ']', '|', '\\', '/', ':', ';', '\"', '\'', '<', '>', ',', '.', '?');
 
 	@PreAuthorize("hasAuthority('Authorities.Client_Password_Policy_CREATE')")
 	@Override
 	public Mono<ClientPasswordPolicy> create(ClientPasswordPolicy entity) {
 
-		return SecurityContextUtil.getUsersContextAuthentication()
+		return flatMapMono(
+
+		        SecurityContextUtil::getUsersContextAuthentication,
+
+		        ca -> entity.getAppId() == null ? this.dao.checkValidEntity(entity.getClientId())
+		                .switchIfEmpty(securityMessageResourceService.throwMessage(
+		                        msg -> new GenericException(HttpStatus.FORBIDDEN, msg),
+		                        SecurityMessageResourceService.MANDATORY_APP_ID, entity.getClientId()))
+		                : Mono.just(true),
 
-				.flatMap(ca -> {
+		        (ca, validEntry) ->
+				{
 
-					ULong currentUser = ULong.valueOf(ca.getLoggedInFromClientId());
+			        ULong currentUser = ULong.valueOf(ca.getLoggedInFromClientId());
 
-					if (ca.isSystemClient() || currentUser.equals(entity.getClientId()))
-						return super.create(entity);
+			        if (ca.isSystemClient() || currentUser.equals(entity.getClientId()))
+				        return super.create(entity);
 
-					return this.clientService.isBeingManagedBy(currentUser, entity.getClientId())
-							.flatMap(managed -> managed.booleanValue() ? super.create(entity) : Mono.empty());
-				})
-				.flatMap(e -> cacheService.evict(CACHE_NAME_CLIENT_PWD_POLICY, e.getClientId())
-						.map(x -> e))
+			        return this.clientService.isBeingManagedBy(currentUser, entity.getClientId())
+			                .flatMap(managed -> managed.booleanValue() ? super.create(entity) : Mono.empty());
+		        }
+
+		).flatMap(e -> cacheService.evict(CACHE_NAME_CLIENT_PWD_POLICY, e.getClientId(), getAppIdIfExists(e.getAppId()))
```

Comment:
The cache policy is wrong. You need to cache against the urlappcode, urlClientCode, and client ID of the user. But not at the app level or client level.